File: python/hyperid/auth/auth_token.py
import time
import jwt 

class AuthToken:
    def __init__(self, token: str):
        self._token = token
    
    def __str__(self):
        return self._token
    
    @property
    def token(self) -> str:
        return self._token

    @token.setter
    def token(self, value: str):
        self._token = value

    def get_decoded_token(self):
        return jwt.decode(self._token, options={"verify_signature": False})

    def is_expired(self) -> bool:
        decoded_token = jwt.decode(self._token, options={"verify_signature": False})
        exp_timestamp = decoded_token.get('exp')
        if not exp_timestamp:
            return True
        current_timestamp = time.time()
        return current_timestamp > exp_timestamp

File: php/hyperid/base/auth_token.php
<?php

class AuthToken {
    public string $token;

    function __construct(string $token) {
        $this->token = $token;
    }

    function setToken(string $token) {
        $this->token = $token;
    }

    function getToken() {
        return $this->token;
    }

    function getDecodedToken() : array {
        if($this->token == null || empty($this->token)) return [];
        list($header, $payload, $signature) = explode('.', $this->token);
        return json_decode(base64_decode(strtr($payload, '-_', '+/')), true);
    }

    function isExpired() : bool {
        if($this->token == null || empty($this->token)) return true;
        $decodedToken = $this->getDecodedToken();
        $exp_timestamp = $decodedToken['exp'];
        if($exp_timestamp === null) return false;
        return time() > $exp_timestamp;
    }
}

?>